File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.py
def factorial(x):
    result = 1
    for i in range(1, x+1):
        result *= i
    return result

def solution(n):
    count = 0
    for i in range(1, n+1):
        tmp = n-i
        count += factorial(i) // ((factorial(i-tmp)) * factorial(tmp))
    return count % 1234567

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int n) {
    vector<int> arr(n, 0);
    arr[0] = 1;
    arr[1] = 2;
    for (int i = 2; i < n; i++) {
        arr[i] = (arr[i-2] + arr[i-1]) % 1234567;
    }
    return arr[n-1];
}